### PR TITLE
Add rtd-poetry

### PR DESCRIPTION
**Pull request by oblute**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by oblute** [comment]
I was previously going to maintain this (https://github.com/conda-forge/staged-recipes/pull/13374) when licensing issues prevented it from going into the forge. Not my original PR, so making a new one now that the developer has packaged a license on github for it.

File: recipes/rtd-poetry/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
+    - intreehooks
```
Comment (dopplershift) [inline recipes/rtd-poetry/meta.yaml:20]: Does this actually need `intreehooks` at package build time?

File: recipes/rtd-poetry/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 21f6d5bc8c2347b1d20c358fcba8f6052a637f3687a1a42393a42e75570ee67c
+
+build:
+  number: 0
```
Comment (dopplershift) [inline recipes/rtd-poetry/meta.yaml:13]: This looks like pure Python? If so, you should be able to add noarch here:
```suggestion
  number: 0
  noarch: python
```

File: recipes/rtd-poetry/meta.yaml (hunk 1 above)
Comment (dopplershift) [reply recipes/rtd-poetry/meta.yaml]: I guess the answer to this question is: "Yes, it does need `intreehooks` at build time." 😉

**Comment by oblute** [comment]
@dopplershift this is ready to go

**Pull request merged**